# Chapter 1: Variables

-------------------------------

## Final Exercises Chapter 1

-  Ex. 1: Suppose the cover price of a book is 24.95 EUR, but bookstores get a 40 percent discount. Shipping costs 3 EUR for the first copy and 75 cents for each additional copy. What is the total wholesale cost for 60 copies? As an extra challenge, try printing the result in a pretty fashion, using casting where necessary!

In [12]:
# your code goes here
book = 24.95 - (24.95/100*40)
shipping_cost_book1 = 3
shipping_cost_more_books = 0.75
copies = 60
total = book*copies + shipping_cost_book1 + shipping_cost_more_books*(copies -1)
print(total)

945.4499999999999


-  Ex. 2: Can you identify and explain the errors in the following lines of code? Correct them please!

In [14]:
print("A message")
print("A message")
print('A messagef')

A message
A message
A messagef


-  Ex. 3: When something is wrong with your code, Python will raise errors. Often these will be 'syntax errors' that signal that something is wrong with the form of your code (i.e. a `SyntaxError` like the one thrown in the previous exercise). There are also 'runtime errors' that signal that your code was in itself formally correct, but that something went wrong during the code's execution. A good example is the `ZeroDivisionError`. Try to make Python throw such a ZeroDivisionError! (Hint: it has to do with the fact that you cannot divide a number by zero)

In [15]:
# ZeroDivisionError
4/0

ZeroDivisionError: division by zero

- Ex. 4: Write a program that assigns the result of `9.5 * 4.5 - 2.5 * 345.5 - 3.5` to a variable. Print this variable. Use round brackets to indicate 'operator precedence' and make sure that subtractions are performed before multiplications. Print the new result. When you convert the outcome to a string, how many characters does it count? Print that too.

In [28]:
# insert your code here
x = 9.5 * 4.5  - 2.5 * 345.5 - 3.5
y = 9.5*(4.5 - 2.5)*(345.5 - 3.5)
print(x)
print(y)
z = len(str(y))
print(z)

-824.5
6498.0
6


-  Ex. 5: Define three variables `var1`, `var2` and `var3`. Calculate the average of these variables and assign it to `average`. Print the result in a fancy manner. Add three comments to this piece of code using three different ways.

In [35]:
# average
var1 = 3 #variablenumberone
var2 = 11 #variablenumbertwo
var3 = 16
average = (var1 + var2 + var3)/3
"When we print, we will see the result"
print("The average of the three variables is", average)


The average of the three variables is 10.0


-  Ex. 6: Write a little program that can compute the surface of a circle, using the variables `radius` and `pi=3.14159`. The formula is of course radius, multiplied by radius, multiplied by `pi`. Print the outcome of your program as follows: 'The surface area of a circle with radius ... is: ...'.

In [39]:
# circle code
radius = 6
pi = 3.14159
x = radius * radius * pi
print("The surface area of a circle with radius", radius, "is", x)

The surface area of a circle with radius 6 is 113.09724


To make things even more interactive, you can ask the user of the program to input the radius. Since this input is a string, you will need to cast it as an integer before doing calculations with it. Try adding your code from above to the code below and see if you can get it to work interactively.

In [41]:
# interactive circle code
radius = input("What is the radius? ")
radius = int(radius) # The interactive input is a string, so we need to cast it as an integer
pi = 3.14159
x = radius * radius * pi
print("The surface area of a circle with radius", radius, "is", x)

What is the radius? 9
The surface area of a circle with radius 9 is 254.46878999999998


- Ex. 7: Get help online. There is one operator (like the ones for multiplication and subtraction) that we did not mention yet, namely the modulus operator `%`. Could you figure by yourself what it does when you place it between two numbers (e.g. 113 % 9)? If not, ask Google for help.

In [49]:
# try out the modulus operator!
x = 100 % 20
print(x)
y = 120 % 6
print(y)
z = 99 % 20
print(z)
a = 101 % 20
print(a)
#The modulus operator divides the first number by the second one. The number you get when you print, is the remainder of the division.

0
0
19
1


- Ex. 8: As happens very often, we make mistakes and Python raises errors. Correct all the errors in the following block of code. 

In [57]:
child1 = 'Bert'
child2 = 'An'
age_child1 = 5
age_child2 = 2

mother = 'Marie'
father = 'Jozef'

message1 = mother + ' and ' + father + ' have two children, ' + child1 + ' and ' + child2 +'.'
message2 = ' Between the two children there is a difference of age of ' + str(age_child1 - age_child2) + ' years.'
print(message1 + message2)


Marie and Jozef have two children, Bert and An. Between the two children there is a difference of age of 3 years.


-----------------------

You've reached the end of Chapter 1! You can safely ignore the code block below -- it's only there to make the page prettier.

In [ ]:
from IPython.core.display import HTML
def css_styling():
    styles = open("styles/custom.css", "r").read()
    return HTML(styles)
css_styling()